src/dharma.ts

```diff
@@ -27,25 +41,27 @@ class Dharma {
 
     public readonly web3: Web3;
 
-    constructor(blockchainHost?: string, addressBook: DharmaTypes.AddressBook = {}) {
+    constructor(web3?: Web3, addressBook: DharmaTypes.AddressBook = {}) {
         /**
-         * There are two ways we can access a web3 provider:
-         * 1. We pass in the address of an Eth node, e.g. https://localhost:8545
-         * 2. Web3 has been injected into the browser window (e.g. via Metamask.)
+         * There are two ways we can access a Web3 instance:
+         * 1. We pass in Web3
+         * 3. Web3 has been injected into the browser window (e.g. via Metamask.)
          */
-        if (blockchainHost) {
-            // If a host is specified, instantiate web3 with it as the provider.
-            const web3Provider = new Web3.providers.HttpProvider(blockchainHost);
-            this.web3 = new Web3(web3Provider);
+        if (web3) {
+            this.verifyWeb3Version(web3);
+
+            this.web3 = web3;
         } else if (
             typeof (window as any) !== "undefined" &&
             typeof (window as any).web3 !== "undefined"
         ) {
-            // If web3 is available via the browser window, instantiate web3 via the current provider.
+            // If Web3 is available via the browser window, instantiate Web3 via the current provider.
             this.web3 = new Web3((window as any).web3.currentProvider);
         } else {
             // Otherwise throw...
-            throw new Error("Pass in the address of your blockchain node.");
+            throw new Error(
+                "A Web3 instance was not specified as a parameter and Web3 was not found on the window.",
```

Comment:
I would consider adding a constant at the top that specifies all the errors thrown.